src/udev/udev-varlink.c

```diff
@@ -0,0 +1,166 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "env-util.h"
+#include "strv.h"
+#include "udev-manager.h"
+#include "udev-varlink.h"
+#include "varlink-io.systemd.Service.h"
+#include "varlink-io.systemd.Udev.h"
+#include "varlink-util.h"
+
+#define UDEV_VARLINK_ADDRESS "/run/udev/io.systemd.Udev"
+
+static int vl_method_reload(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        assert(link);
+
+        if (sd_json_variant_elements(parameters) > 0)
+                return sd_varlink_error_invalid_parameter(link, parameters);
+
+        log_debug("Received io.systemd.Service.Reload");
+        manager_reload(userdata, /* force = */ true);
+        return sd_varlink_reply(link, NULL);
+}
+
+static int vl_method_set_log_level(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        int r, level;
+
+        assert(link);
+
+        r = varlink_dispatch_set_log_level(link, parameters, &level);
```

Comment:
i sense that sooner or later we'll have multiple places where we want to dispatch a syslog log level from json into a c struct, hence maybe implement a generic json_dispatch_log_level() or so in src/libsystemd/sd-json/json-util.[ch] that can do this properly, similar to how we have json_dispatch_ifindex() there and similar helpers.